src/lib.rs

```diff
@@ -445,13 +447,13 @@ impl<Types: IpfsTypes> Ipfs<Types> {
         self.ipns().cancel(key).instrument(self.span.clone()).await
     }
 
-    pub async fn connect<T: Into<ConnectionTarget>>(&self, target: T) -> Result<(), Error> {
+    pub async fn connect(&self, target: Multiaddr) -> Result<(), Error> {
         self.span
```

Comment:
should here be a check for target ending in the p2p part or am I missing something?

Reply:
Note this `.any(..)` check isn't good enough but lets fix it in the follow up.